### DISCUSSION TITLE
Center simplegrid on Chakra UI

### DISCUSSION THREAD

**Discussion by CodeDruid13** [chakra-ui/chakra-ui#7389] [Q&A]
I have an array of items but I'm trying to center the items without affecting columns. Using display={"flex"} and flexWrap={"center"} centers the items but affects the number of columns I can display. Removing the display and flexwrap aligns the items to start.

     <SimpleGrid
                      mb={[5, 10, 10]}
                      spacing={[5, 10, 10]}
                      columns={[1, 1, 2, 4]}
                      mt="15px"
                      justifyContent={"center"}
                      alignContent={"center"}
                      display={"flex"}
                      flexWrap={"wrap"}
                    >
                      {items?.map((item, idx) => (
                        <Box pos="relative" p={6} key={idx} rounded="full">
                          <Skeleton
                            h="175px"
                            w={"175px"}
                            mt={"10px"}
                            rounded={"full"}
                            fadeDuration={4}
                            startColor={"gray.50"}
                            endColor={"gray.200"}
                          />
    
                          <Skeleton
                            h="10px"
                            mt={"10px"}
                            rounded={"full"}
                            fadeDuration={4}
                            startColor={"gray.50"}
                            endColor={"gray.200"}
                          />
    
                          <Skeleton
                            h="10px"
                            mt={"10px"}
                            rounded={"full"}
                            fadeDuration={4}
                            startColor={"gray.50"}
                            endColor={"gray.200"}
                          />
                        </Box>
                      ))}
                    </SimpleGrid>

The above sample code produces [![Centered Images][1]][1] but affecting the columns set to 4. Removing display and flexWrap produces [![Aligned Images][2]][2] and gives 4 columns but aligned to start. 
Adding align and justify to center does not seem to work either. Anyone with an idea on how to achieve this in an easier way?

  [1]: https://i.stack.imgur.com/Bnm8Q.png
  [2]: https://i.stack.imgur.com/V6g3J.png

**Comment by keithburgie** [comment]
Any chance you could provide an image of what you're trying to create? I'm not able to follow.

**Reply by CodeDruid13** [reply]
> Any chance you could provide an image of what you're trying to create? I'm not able to follow.

This is what am trying to achieve [![Centered Images][1]][1]. Using FlexWrap and display to flex works but that ignores number of columns I want to use. The items for the simplegrid are dynamic and I wanted to center the last row.

  [1]: https://i.stack.imgur.com/Bnm8Q.png

**Comment by keithburgie** [comment]
CSS Grid isn't going to let you do that.

TLDR: I made you some examples that show how you might achieve your layout with Flex or Grid. If you're using an uncertain number of items, it gets a lot more complicated, but you can take a look here and try changing some of the numbers and properties: https://codesandbox.io/s/small-water-l26l96?file=/App.tsx

With your code:
1. Using `Grid` or `SimpleGrid` is like using `<Box display="grid">`
2. Using `Flex` or `Stack` is like using `<Box display="flex">`

By using `<SimpleGrid display="flex">` you are turning example 1 into example 2. You're un-doing `display: grid`.

From there, you are using `items.map()` to return an entire row that has three children inside it. To properly use `SimpleGrid`, it would need to look more like the code below. Children of a "grid" component follow the rules of their parent.
```
const items = [1, 2, 3, 4, 5];

return (
  <SimpleGrid columns={3} spacing="20px" p={2}>
    {items.map((item) => (
      <Center key={item} bg="tomato" height="80px">
        {item}
      </Center>
    ))}
  </SimpleGrid>
);
```
<img width="506" alt="Screenshot 2023-02-24 at 1 30 15 PM" src="https://user-images.githubusercontent.com/96078669/221261102-fc26e7e1-54d2-4d9d-ba69-a6aa04794fd1.png">
